zt_backend/router.py

```diff
@@ -610,89 +636,150 @@ def create_item(item: request.CreateItemRequest):
 
 
 @router.post("/api/rename_item")
-def rename_item(rename_request: request.RenameItemRequest):
-    if app_state.run_mode == "dev":
-        try:
-            old_path = Path(rename_request.path) / rename_request.oldName
-            new_path = Path(rename_request.path) / rename_request.newName.strip()
-
-            if not old_path.exists():
-                raise HTTPException(status_code=404, detail="Item not found")
-
-            if old_path == new_path:
-                return {
-                    "success": True,
-                    "message": f"Item renamed successfully (no change in name)",
-                    "oldPath": str(old_path),
-                    "newPath": str(new_path),
-                }
-
-            if new_path.exists():
-                raise HTTPException(
-                    status_code=400, detail="An item with the new name already exists"
-                )
-
-            os.rename(old_path, new_path)
-
-            return {
-                "success": True,
-                "message": f"Item renamed successfully from {rename_request.oldName} to {rename_request.newName}",
-                "oldPath": str(old_path),
-                "newPath": str(new_path),
-            }
-        except PermissionError:
-            raise HTTPException(
-                status_code=403, detail="Permission denied. Unable to rename the item."
-            )
-        except OSError as e:
-            raise HTTPException(status_code=500, detail=f"System error: {str(e)}")
-        except Exception as e:
-            raise HTTPException(
-                status_code=500, detail=f"An unexpected error occurred: {str(e)}"
-            )
-
-
+async def rename_item(request: request.RenameItemRequest):
+    try:
```

Comment:
we don't want to remove the if dev mode here. Otherwise this endpoint is available on published apps